Translate attribute label with default translation helper function

cmuench opened the pull request:
Adds the translation function to customer attribute labels in Magento
admin. This gives a chance to translate a label in the locale of a
backend user.

<!--- Provide a general summary of the Pull Request in the Title above -->

### Description
A custom attribute is created.
We have different frontend labels for the attribute. Store views with different locale settings exist.
Admin users are speaking different languages and need a localized attribute label.
This is not possible.


### Fixed Issues (if relevant)
<!--- Provide a list of fixed issues in the format magento/magento2#<issue_number>, if relevant  -->
1. magento/magento2#13251: Unable to translate custom attributes in admin backend

### Manual testing scenarios
1. Create an custom attribute with different frontend labels.

### Contribution checklist
 - [X] Pull request has a meaningful description of its purpose
 - [X] All commits are accompanied by meaningful commit messages
 - [X] All new or changed code is covered with unit/integration tests (if applicable)
 - [X] All automated tests passed successfully (all builds on Travis CI are green)

cmuench committed:
Translate attribute label with default translation helper function

Adds the translation function to customer attribute labels in Magento
admin. This gives a chance to translate a label in the locale of a
backend user.

miguelbalparda commented:
@cmuench can you check the build please? It seems `new Phrase('')` is the line causing them to fail.

cmuench commented:
@miguelbalparda Yes, I will check this.

cmuench commented:
@miguelbalparda I checked the failing test. It is indirectly caused by my change. Sorry for the hard words... The failing test is totally s****.
The test mocks components. Then in the test setup a generator is created by calling `ServiceLocator::getDictionaryGenerator()`. This loads ALL components of the system. In my case >12.000 files are scanned. This slows down the complete test execution and creates a lot of IOs on my disk.
I try to fix this test.

cmuench committed:
Fix for missing context reset of ComponentRegistrar

This test tries to setup a test environment with two modules and one
theme to test the generator.
During the test run, the test environment and the complete module stack
is processed by the generator.
So the test takes on my machine 25 seconds with xdebug enabled.
After resetting the registrar the complete test runs in ~50ms with
xdebug enabled.
Without the reset ~12.000 files are scanned. With the reset only 5 files
are scanned.

This test is not a unit test. It is an integration test.

cmuench commented:
@miguelbalparda I "fixed" the test not my code. Now the failing test runs in ~30ms instead of 25 seconds. My patch will speed up the complete test environment.

miguelbalparda commented:
Please don't insult other's people work. I've edited your comment. Thanks for the fix, we'll check it and report back.

cmuench commented:
@miguelbalparda You should know me. I never insult any developers work. I only hate the written code. Every developer should hate the own written code and improve it. With this code I had some issues in a project.

magento-engcom-team commented:
@cmuench thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

cmuench commented:
@magento-engcom-team Thanks but the link is always redirecting me to https://github.com/magento/

duckchip commented:
Reproduced on Magento 2.2.5

vitmarks commented:
Hello, I wanted to ask, whether this issue was fixed?
I still have this problem. I'am using magento 2.3 beta 2
and I created some attributes with different labels, but If I switch to another backend user with configurated language FR, I still see all my attributes as Default labeled.

Did I miss something?
If there is not a fix til now, is there a quick fix (workaround), that I can implement for now? 

thx
